api/python/quilt3/admin/__init__.py

```diff
@@ -0,0 +1,178 @@
+"""
+APIs for Quilt administrators. 'Registry' refers to Quilt stack backend services, including identity management.
+"""
+# This wraps code generated by aridne-codegen to provide a more user-friendly API.
+from datetime import datetime
+from typing import Annotated, Any, List, Literal, Optional, Union
+
+import pydantic
+
+from . import _graphql_client
+
+
+@pydantic.dataclasses.dataclass
+class ManagedRole:
+    id: str
+    name: str
+    arn: str
+    typename__: Literal["ManagedRole"]
+
+
+
+@pydantic.dataclasses.dataclass
+class UnmanagedRole:
+    id: str
+    name: str
+    arn: str
+    typename__: Literal["UnmanagedRole"]
+
+
+Role = Union[ManagedRole, UnmanagedRole]
+AnnotatedRole = Annotated[Role, pydantic.Field(discriminator="typename__")]
+role_adapter = pydantic.TypeAdapter(AnnotatedRole)
+
+
+@pydantic.dataclasses.dataclass
+class User:
+    name: str
+    email: str
+    date_joined: datetime
+    last_login: datetime
+    is_active: bool
+    is_admin: bool
+    is_sso_only: bool
+    is_service: bool
+    role: Optional[AnnotatedRole]
+    extra_roles: List[AnnotatedRole]
+
+
+class Quilt3AdminError(Exception):
+    def __init__(self, details):
+        super().__init__(details)
+        self.details = details
+
+
+class UserNotFoundError(Quilt3AdminError):
+    def __init__(self):
+        super().__init__(None)
+
+
+def _handle_errors(result: _graphql_client.BaseModel) -> Any:
+    if isinstance(result, (_graphql_client.InvalidInputSelection, _graphql_client.OperationErrorSelection)):
+        raise Quilt3AdminError(result)
+    return result
+
+
+def _get_client():
+    return _graphql_client.Client()
+
+
+def get_user(name: str) -> Optional[User]:
+    """
+    Get a specific user from the registry. Return `None` if the user does not exist.
+
+    Args:
+        name: Username of user to get.
+    """
+    result = _get_client().get_user(name=name)
+    if result is None:
+        return None
+    return User(**result.model_dump())
+
+
+def get_users() -> List[User]:
+    """
+    Get a list of all users in the registry.
+    """
+    return [User(**u.model_dump()) for u in _get_client().get_users()]
+
+
+def create_user(name: str, email: str, role: str, extra_roles: Optional[List[str]] = None) -> None:
```

Comment:
shouldn't we return the created user data here?

Reply:
I'm not sure that's really useful

Reply:
there is data not directly inferrable from the input params (generated/inferred by the backend), like isAdmin flag and such -- tho i'm not sure it's that useful either